frontend/src/views/Game.tsx

```diff
@@ -1,13 +1,44 @@
 import Editor from '@monaco-editor/react';
 import React from 'react';
+import { Problem, getProblems } from '../api/problem';
+import styles from './styles.module.css';
 
-function GamePage() {
-  return (
-    <div>
-      Game page
-      <Editor height="90vh" language="javascript" />
-    </div>
-  );
+interface GameState {
+  problems?: Problem[];
+}
+
+interface GameProps {}
+
+class GamePage extends React.Component<GameProps, GameState> {
```

Comment:
I'm curious on @alankbi's thoughts on this; I was under the impression that we were going to use functions, and incorporate the `state` variable similar to [this React tutorial](https://reactjs.org/docs/hooks-state.html). I'm assuming you changed it to a class when seeing [the "State and Lifecycle" React tutorial](https://reactjs.org/docs/state-and-lifecycle.html) -- this may be better I'm just not sure, do you guys know what the pros / cons are?

Reply:
The main difference between `useState` in a function and `setState` in a class is that `setState` allows you to define multiple state variables at once while `useState` can only have one. This purpose isn't really expressed in my code as it is because I'm using only one variable which specifies whether problems exist to be displayed. The `componentDidMount()` function in a class can be incredibly useful -- it allows you to wait until other components are rendered before running a certain piece of code. I think I can transition this `GamePage` class into a function if you guys think it's better, I can probably even get rid of `componentDidMount()`,  however I don't think we should necessarily try to avoid using classes for the sake of consistency.

Reply:
Yeah so I think React Hooks (the functional components + useXXX functions) are newer and the ["future"](https://reactjs.org/docs/hooks-faq.html#should-i-use-hooks-classes-or-a-mix-of-both) of React, but were only released recently and class components are still more common in existing apps/tutorials. For the sake of consistency and long-term prospects, it might be good to stick with functional components, but I think the difference isn't too important and we can always convert them later.

Reply:
I also want to add that I think `useState` allows you to set multiple state variables; I did so in PR #26. As for converting them later @alankbi, personally I don't think it would make sense to change it now just to change it back later -- whichever we choose, I think we should use now, as it'd make development much easier.

Reply:
Sounds good, I am switching the class into a function.